pkg/pillar/types/types.go

```diff
@@ -279,6 +279,7 @@ type UEvent struct {
 type UUIDPairToNum struct {
 	BaseID      uuid.UUID
 	AppID       uuid.UUID
+	IfIdx       uint32
```

Comment:
1. Is this fix required to solve the critical problem?
2. Note that this is now completely misnamed since it is uuidpairplusint to a number. If we need to fix this (hopefully separate from the critical problem), then I think it should be a InterfaceToNum where we feed in an Interface {} (which in this case will be two UUIDs plus an integer) and use the Equal function inside the package.